### PR TITLE
Improving support of `counts` and `sample` with `default.mixed`

### PR DESCRIPTION
**Pull request by obliviateandsurrender**
**Context:** `Counts` should work with interfaces (torch, tf, jax) and `default.mixed` but it is not the case. This PR fixes it.

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:** #2984

### CODE REVIEW

**Comment by obliviateandsurrender** [comment]
[sc-25027]

File: tests/devices/test_default_mixed_torch.py [hunk 1]
```diff
@@ -718,3 +718,31 @@ def circuit(weights):
...
+class TestMeasurements:
+    """Tests for measurements with default.mixed"""
+
+    @pytest.mark.parametrize(
+        "measurement",
+        [
+            qml.counts(qml.PauliZ(0)),
+            qml.counts(wires=[0]),
+            qml.sample(qml.PauliX(0)),
+            qml.sample(wires=[1]),
```
Comment (trbromley) [inline tests/devices/test_default_mixed_torch.py:732]: Nice! Might be worth updating the title to mention that `sample` is being supported too :rocket:

**Review by mudit2812 — APPROVED** [review]
LGTM!

**Review by Jaybsoni — APPROVED** [review]
Looks good 👍🏼, 

This might also natively support jax now. If that's the case it would be nice to also add jax tests. Not blocking though.

**Pull request merged**